crates/polars-core/src/chunked_array/ops/split.rs

```diff
@@ -0,0 +1,47 @@
+pub struct SplitNChars<'a> {
+    s: &'a str,
+    n: usize,
+    keep_remainder: bool,
+}
+
+impl<'a> Iterator for SplitNChars<'a> {
+    type Item = &'a str;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let single_char_limit = if self.keep_remainder { 2 } else { 1 };
+        if self.n >= single_char_limit {
+            self.n -= 1;
+            let ch = self.s.chars().next()?;
+            let first;
+            (first, self.s) = self.s.split_at(ch.len_utf8());
+            Some(first)
+        } else if self.n == 1 && !self.s.is_empty() {
```

Comment:
this return remaining string when n == 1, but doesn't check the state of keep_remainder (true/false)